streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -186,57 +205,25 @@ public void cleanup() {
     public void shouldRecordRecordsAndBytesProduced() {
         final Headers headers = new RecordHeaders(new Header[]{new RecordHeader("key", "value".getBytes())});
 
-        final String threadId = Thread.currentThread().getName();
-        final String processorNodeId = sinkNodeName;
-        final String topic = "topic";
-        final Metric recordsProduced = streamsMetrics.metrics().get(
-            new MetricName("records-produced-total",
-                           TOPIC_LEVEL_GROUP,
-                           "The total number of records produced from this topic",
-                           streamsMetrics.topicLevelTagMap(threadId, taskId.toString(), processorNodeId, topic))
-        );
-        final Metric bytesProduced = streamsMetrics.metrics().get(
-            new MetricName("bytes-produced-total",
-                           TOPIC_LEVEL_GROUP,
-                           "The total number of bytes produced from this topic",
-                           streamsMetrics.topicLevelTagMap(threadId, taskId.toString(), processorNodeId, topic))
-        );
-
-        double totalRecords = 0D;
-        double totalBytes = 0D;
+        final MockedStatic<TopicMetrics> topicMetrics = mockStatic(TopicMetrics.class);
 
-        assertThat(recordsProduced.metricValue(), equalTo(totalRecords));
-        assertThat(bytesProduced.metricValue(), equalTo(totalBytes));
+        when(TopicMetrics.producedSensor(
+                Mockito.anyString(),
+                Mockito.anyString(),
+                Mockito.anyString(),
+                Mockito.anyString(),
+                Mockito.any(StreamsMetricsImpl.class)
+        )).thenReturn(mockSensor);
 
         collector.send(topic, "999", "0", null, 0, null, stringSerializer, stringSerializer, sinkNodeName, context);
-        ++totalRecords;
-        totalBytes += producerRecordSizeInBytes(mockProducer.history().get(0));
-        assertThat(recordsProduced.metricValue(), equalTo(totalRecords));
-        assertThat(bytesProduced.metricValue(), equalTo(totalBytes));
+        double bytes = producerRecordSizeInBytes(mockProducer.history().get(0));
```

Comment:
I realized now that my suggestion in my previous review did make too much sense for some parts.
```suggestion
        final double bytesWithoutHeaders = producerRecordSizeInBytes(mockProducer.history().get(0));
```